hitman-vscode integration

mrjackv opened the pull request.

mrjackv committed:
vscode_extension.lib: silence cmp

mrjackv committed:
revng-vscode-web: add profile configuration

Add default profile configuration for revng-vscode-web to allow
access to revng daemon on the default port

mrjackv committed:
revng-ui is now revng

mrjackv committed:
revng-distributable: fix missing env variable

Fix the absence of the `INSTALL_LINK_ONLY_PATH` environment variable
which would cause failure when running revng-distributable. Moreover
change the way revng-distributable's test is run to avoid environment
variable pollution as much as possible

aleclearmind commented on .orchestra/support/vscode-web/product.json:
```diff
@@ -15,7 +15,14 @@
+        "workbench.colorTheme": "revng",
+        "revng.general.profiles": [
+            {
+                "id": "9b473245-159e-4984-a052-f2d9131314ea",
+                "name": "Local",
+                "directURLs": ["http://127.0.0.1:8000/"]
```
You sure this belongs to orchestra?

mrjackv replied:
The only other place to put this would be in `revng-ui` as the default for the options, but I don't know if it's a sensible thing to do (since this is the config for the `revng web-ui` command)

aleclearmind replied:
Uhm OK.

mrjackv committed:
revng-distributable: exclude node_cache

With the inclusion of revng-js-prerequisites as a dependency of revng
the folder `node_cache` is erroneously imported into
revng-distributable. This commit modifies the `install.sh` script to
exclude this folder

aleclearmind commented:
```
s|root/share/vscode-web.tar.xz|root/share/vscode-web-vanilla.tar.xz|
```

mrjackv committed:
vscode-web: change output filename

Change the output filename of the `vscode-web` component to better
reflect that it is the unadulterated version of vscode compiled for the
web target

mrjackv commented:
> ```
> s|root/share/vscode-web.tar.xz|root/share/vscode-web-vanilla.tar.xz|
> ```

Done

mrjackv committed:
vscode-web: import native module dependencies

Import the dependencies needed to successfully build the `native-keymap`
module

mrjackv committed:
fetch-js-dependencies: handle yarn overrides

When yarn overrides a dependency it leaves the original dependency
present with a version of "0.0.0". This commit adds this logic to the
fetch-js-dependencies script

mrjackv committed:
install-system-dependencies.sh: drop old deps

Drop dependencies from `install-system-dependencies.sh` as some were
removed from the codebase and others were added within orchestra in the
meantime
